ws/DiscordWebsocket.ts

```diff
@@ -161,11 +161,11 @@ export default class DiscordWebsocket {
       case GATEWAY_CLOSE_EVENT_CODES.UNKNOWN_ERROR:
       case GATEWAY_CLOSE_EVENT_CODES.UNKNOW_OPCODE:
       case GATEWAY_CLOSE_EVENT_CODES.DECODE_ERROR:
-      case GATEWAY_CLOSE_EVENT_CODES.ALREADY_AUTHENTICATED:
+      case GATEWAY_CLOSE_EVENT_CODES.ALREADY_AUTHENTICATED: this.sessionID = null; break;
       case GATEWAY_CLOSE_EVENT_CODES.RATE_LIMITED: this.restart(); break;
-      case GATEWAY_CLOSE_EVENT_CODES.NOT_AUTHENTICATED:
+      case GATEWAY_CLOSE_EVENT_CODES.NOT_AUTHENTICATED: this.sessionID = null; break;
       case GATEWAY_CLOSE_EVENT_CODES.INVALID_SESSION:
-      case GATEWAY_CLOSE_EVENT_CODES.INVALID_RESUME_SEQUENCE:
+      case GATEWAY_CLOSE_EVENT_CODES.INVALID_RESUME_SEQUENCE: this._seq = null; break;
```

Comment:
slightly confused on these 3. Why would we need these special case?

Reply:
INVALID_RESUME_SEQUENCE is when the seq is incorrect, so we have to reset it, no?

Reply:
They're all handled by restart with true param

Reply:
Alright I used that instead